File: CS/Form1.cs
using System.Collections.Generic;
using System.IO;
using System.Windows.Forms;
using DevExpress.DashboardWin;
using DevExpress.XtraGrid;
using DevExpress.XtraGrid.Views.Base;
using DevExpress.XtraGrid.Views.Grid;

namespace DataInspectorCustomizationExample {
    public partial class Form1 : DevExpress.XtraEditors.XtraForm
    {
        Dictionary<string, Stream> aggregatedGridLayoutDictionary = new Dictionary<string, Stream>();
        Dictionary<string, Stream> rawGridLayoutDictionary = new Dictionary<string, Stream>();
        public Form1() {
            InitializeComponent();
            dashboardViewer1.AllowInspectAggregatedData = true;
            dashboardViewer1.AllowInspectRawData = true;
            dashboardViewer1.DataInspectorFormLoad += DashboardViewer1_DataInspectorFormLoad;
            dashboardViewer1.DataInspectorFormClosing += DashboardViewer1_DataInspectorFormClosing;
        }

        private void DashboardViewer1_DataInspectorFormClosing(object sender, DataInspectorFormClosingEventArgs e) {
            SaveGridLayout(e.DashboardItemName, e.AggregatedGrid.FocusedView, aggregatedGridLayoutDictionary);
            SaveGridLayout(e.DashboardItemName, e.RawGrid.FocusedView, rawGridLayoutDictionary);
        }
        private void DashboardViewer1_DataInspectorFormLoad(object sender, DataInspectorFormLoadEventArgs e) {
            CustomizeGrid(e.AggregatedGrid);
            CustomizeGrid(e.RawGrid);

            RestoreGridLayout(e.DashboardItemName, e.AggregatedGrid.FocusedView, aggregatedGridLayoutDictionary);
            RestoreGridLayout(e.DashboardItemName, e.RawGrid.FocusedView, rawGridLayoutDictionary);
        }
        void CustomizeGrid(GridControl grid) {
            GridView view = grid.MainView as GridView;
            view.OptionsCustomization.AllowGroup = true;
            view.OptionsView.ShowGroupPanel = true;
            view.OptionsBehavior.AutoExpandAllGroups = true;
        }
        void SaveGridLayout(string itemName, BaseView gridView, Dictionary<string, Stream> layouts) {
            MemoryStream str = new MemoryStream();
            gridView.SaveLayoutToStream(str);
            layouts[itemName] = str;
            str.Seek(0, SeekOrigin.Begin);
        }
        void RestoreGridLayout(string itemName, BaseView gridView, Dictionary<string, Stream> layouts) {
            Stream stream = null;
            if(layouts.TryGetValue(itemName, out stream)) {
                gridView.RestoreLayoutFromStream(stream);
                stream.Seek(0, System.IO.SeekOrigin.Begin);
            }
        }
    }
}

File: VB/Form1.vb
Imports System.Collections.Generic
Imports System.IO
Imports System.Windows.Forms
Imports DevExpress.DashboardWin
Imports DevExpress.XtraGrid
Imports DevExpress.XtraGrid.Views.Base
Imports DevExpress.XtraGrid.Views.Grid

Namespace DataInspectorCustomizationExample
	Partial Public Class Form1
		Inherits DevExpress.XtraEditors.XtraForm

		Private aggregatedGridLayoutDictionary As New Dictionary(Of String, Stream)()
		Private rawGridLayoutDictionary As New Dictionary(Of String, Stream)()
		Public Sub New()
			InitializeComponent()
			dashboardViewer1.AllowInspectAggregatedData = True
			dashboardViewer1.AllowInspectRawData = True
			AddHandler dashboardViewer1.DataInspectorFormLoad, AddressOf DashboardViewer1_DataInspectorFormLoad
			AddHandler dashboardViewer1.DataInspectorFormClosing, AddressOf DashboardViewer1_DataInspectorFormClosing
		End Sub

		Private Sub DashboardViewer1_DataInspectorFormClosing(ByVal sender As Object, ByVal e As DataInspectorFormClosingEventArgs)
			SaveGridLayout(e.DashboardItemName, e.AggregatedGrid.FocusedView, aggregatedGridLayoutDictionary)
			SaveGridLayout(e.DashboardItemName, e.RawGrid.FocusedView, rawGridLayoutDictionary)
		End Sub
		Private Sub DashboardViewer1_DataInspectorFormLoad(ByVal sender As Object, ByVal e As DataInspectorFormLoadEventArgs)
			CustomizeGrid(e.AggregatedGrid)
			CustomizeGrid(e.RawGrid)

			RestoreGridLayout(e.DashboardItemName, e.AggregatedGrid.FocusedView, aggregatedGridLayoutDictionary)
			RestoreGridLayout(e.DashboardItemName, e.RawGrid.FocusedView, rawGridLayoutDictionary)
		End Sub
		Private Sub CustomizeGrid(ByVal grid As GridControl)
			Dim view As GridView = TryCast(grid.MainView, GridView)
			view.OptionsCustomization.AllowGroup = True
			view.OptionsView.ShowGroupPanel = True
			view.OptionsBehavior.AutoExpandAllGroups = True
		End Sub
		Private Sub SaveGridLayout(ByVal itemName As String, ByVal gridView As BaseView, ByVal layouts As Dictionary(Of String, Stream))
			Dim str As New MemoryStream()
			gridView.SaveLayoutToStream(str)
			layouts(itemName) = str
			str.Seek(0, SeekOrigin.Begin)
		End Sub
		Private Sub RestoreGridLayout(ByVal itemName As String, ByVal gridView As BaseView, ByVal layouts As Dictionary(Of String, Stream))
			Dim stream As Stream = Nothing
			If layouts.TryGetValue(itemName, stream) Then
				gridView.RestoreLayoutFromStream(stream)
				stream.Seek(0, System.IO.SeekOrigin.Begin)
			End If
		End Sub
	End Class
End Namespace
